_posts/2021-05-27-TC-Guard.md

```diff
@@ -0,0 +1,160 @@
+When the application needs this data, it is retrieved again via the TC Guard API.
+The last block connected to the TC Guard API is Prometheus.
+Prometheus is used to observe the API.
+The resulting data is then displayed in the Grafana dashboard.
+
+### Argo CD
+Argo CD is a Continuous Delivery GitOps tool for Kubernetes.
+Within the internship assignment, this is used to install helm charts on the Kubernetes cluster.
+If a change is made on the Azure DevOps repository, Argo CD will detect this and implement the changes on the Kubernetes cluster via a rolling update.
+
+It was quite satisfying to observe the changes being applied.
+Argo CD made managing our applications way easier.
+For example when we updated Graylog the only thing we had to do was change the version number in a .yaml file and Argo CD took care of everything else.
+
+<img src = "/img/2021-05-27-TC-Guard/argocd.png" class = "image fit">
+
+### Ingress
+Ingress provides routing rules to manage remote users' access to services in a Kubernetes cluster. With Ingress, simple rules can be set for routing traffic without exposing every service on the endpoint.
+While setting up Ingress, a few issues surfaced due to our inexperience with Kubernetes.
+Because the online documentation of Ingress nginx was rather limited, setting up Ingress took a fair amount of time.
+Afterwards, Ingress nginx was replaced by Contour, by this time we had more experience.
+Because of this installation went smoothy.
+
+### Graylog
+The centralized logging system aims to obtain the logs of the API and the Android application in a central place so that it would be easier to detect errors.
+To obtain and view the logs in a central place, Graylog is used.
+We chose Graylog because it is much easier to use than Kibana.
+We knew that Kibana has more features than Graylog, but for this project it was not required.
+
+<img src = "/img/2021-05-27-TC-Guard/graylog.png" class = "image fit">
+
+### Grafana
+Prometheus was used to observe the API.
+From this, certain metrics were displayed in the Grafana dashboard.
+Prometheus makes use of targets.
+On this target Prometheus will collect its metrics.
+To add a target the config file of Prometheus had to be modified.
+Because we were not sure how to modify the config file, a lot of time was lost.
+To solve this problem we had to create a secret that contains the target.
+This secret should then be added to the prometheus.yaml and then the target was added.
+
+<img src = "/img/2021-05-27-TC-Guard/grafana.png" class = "image fit">
+
+# Conclusion
+During this internship there were a lot of new technologies covered such as Spring, Kubernetes, Graylog and so on.
+The internship assignment consisted of three main parts including the Android application, the Spring Boot API and Kubernetes.
+Now we are going to talk about how we have experienced the learning curve for each component.
+First, we are going to talk about the Android Application.
```

Comment:
don't announce what you are going to talk about, just telling what you want to say.